pymatgen/io/vasp/inputs.py

```diff
@@ -122,12 +122,22 @@ def __init__(
         """
         if structure.is_ordered:
             site_properties = {}
-            if selective_dynamics:
-                site_properties["selective_dynamics"] = selective_dynamics
-            if velocities:
-                site_properties["velocities"] = velocities
-            if predictor_corrector:
-                site_properties["predictor_corrector"] = predictor_corrector
+
+            if selective_dynamics is not None:
+                selective_dynamics = np.array(selective_dynamics)
+                if not selective_dynamics.all():
+                    site_properties["selective_dynamics"] = selective_dynamics
```

Comment:
@chiang-yuan I simplified the code some. Was there a reason to convert `np.array` to list before assigning to `site_props`?

Reply:
@janosh I converted np.array back to list because the other unit test in `io/vasp/tests/test_inputs.py` compare `selective_dyanmics` with list, which will yield the error in pytest like the one after your revise. 

```
>           assert poscar.selective_dynamics == [[True, True, True], [False, False, False]]
E           ValueError: The truth value of an array with more than one element is ambiguous. Use a.any() or a.all()

/home/runner/work/pymatgen/pymatgen/pymatgen/io/vasp/tests/test_inputs.py:87: ValueError
```

I bypass them by converting it back to list (which is what the original code intended to do) and adding another unit test comparing by list as well. Do you think it is better to revise the previous unit test and compare all of them in the format of `nd.array` like I suggested earlier?

Reply:
I think we should make the test datatype agnostic so that it handles both arrays and lists. It's better for tests to care only about semantics and not equivalent ways of encoding the same meaning.

Reply:
Agree. Thanks @janosh !